plugin-server/src/worker/ingestion/person-state.ts

```diff
@@ -751,56 +771,72 @@ export class DeferredPersonOverrideWriter {
+                        (limit !== undefined ? ` LIMIT ${limit}` : ''),
                     undefined,
                     'processPendingOverrides'
                 )
+
+                const messages: ProducerRecord[] = []
+                for (const { id, ...mergeOperation } of rows) {
+                    messages.push(...(await this.writer.addPersonOverride(tx, mergeOperation)))
+                    await this.postgres.query(
+                        tx,
+                        SQL`DELETE FROM posthog_pendingpersonoverride WHERE id = ${id}`,
+                        undefined,
+                        'processPendingOverrides'
+                    )
+                }
+
+                // n.b.: We publish the messages here (and wait for acks) to ensure
+                // that all of our override updates are sent to Kafka prior to
+                // committing the transaction. If we're unable to publish, we should
+                // discard updates and try again later when it's available -- not
+                // doing so would cause the copy of this data in ClickHouse to
+                // slowly drift out of sync with the copy in Postgres. This write is
+                // safe to retry if we write to Kafka but then fail to commit to
+                // Postgres for some reason -- the same row state should be
+                // generated each call, and the receiving ReplacingMergeTree will
+                // ensure we keep only the latest version after all writes settle.)
+                await this.kafkaProducer.queueMessages(messages, true)
+
+                return rows.length
             }
+        )
 
-            // n.b.: We publish the messages here (and wait for acks) to ensure
-            // that all of our override updates are sent to Kafka prior to
-            // committing the transaction. If we're unable to publish, we should
-            // discard updates and try again later when it's available -- not
-            // doing so would cause the copy of this data in ClickHouse to
-            // slowly drift out of sync with the copy in Postgres. This write is
-            // safe to retry if we write to Kafka but then fail to commit to
-            // Postgres for some reason -- the same row state should be
-            // generated each call, and the receiving ReplacingMergeTree will
-            // ensure we keep only the latest version after all writes settle.)
-            await kafkaProducer.queueMessages(messages, true)
-
-            return rows.length
-        })
+        deferredPersonOverridesProcessedCounter.inc(overridesCount)
+
+        return overridesCount
+    }
+
+    public runTask(intervalMs: number): PeriodicTask {
+        return new PeriodicTask(async () => {
+            status.debug('👥', 'Processing pending overrides...')
+            const overridesCount = await instrument({ metricName: 'processPendingOverrides' }, () =>
+                this.processPendingOverrides()
+            )
```

Comment:
Seems like this `instrument` might be generically useful enough to just move into `PeriodicTask`?

Reply:
Yeah, probably.

Reply:
Cool, added this in [b7de72b](https://github.com/PostHog/posthog/pull/19112/commits/b7de72b7c802d604f12e1853a2d7301a336fe36e).